### ISSUE TITLE
2.6-202412 Public Comment Feedback

### ISSUE THREAD

**Issue by hillslatt** [InteractiveAdvertisingBureau/openrtb2.x#121]
Please comment on this issue with feedback on updates to the OpenRTB updates in Public Comment

OpenRTB Updates:
- Addition of `gtax` and `genres` attributes in [Object: Content](https://github.com/InteractiveAdvertisingBureau/openrtb2.x/blob/develop/2.6.md#3216---object-content-)
- Section in [Implementation Guidance](https://github.com/InteractiveAdvertisingBureau/openrtb2.x/blob/develop/implementation.md#713---using-genres-and-gtax-attributes-) on using new attributes.
- Updated AdCOM [List: Category Taxonomies](https://github.com/InteractiveAdvertisingBureau/AdCOM/blob/develop/AdCOM%20v1.0%20FINAL.md#list--category-taxonomies-) with enum for Content Taxonomy 3.1

Taxonomy Updates:
- Updated version of [Content Taxonomy 3.1](https://github.com/InteractiveAdvertisingBureau/Taxonomies/blob/develop/Content%20Taxonomies/Content%20Taxonomy%203.1.tsv) to add row for Horror
- [Sub-set of Content Taxonomy 3.1](https://github.com/InteractiveAdvertisingBureau/Taxonomies/blob/develop/Taxonomy%20Mappings/CTV%20Genre%20Mapping.tsv) with expected values for `genre`

**Comment by patmmccann** [comment]
If you open the PR, https://github.com/InteractiveAdvertisingBureau/openrtb2.x/compare/main...develop, it is a lot easier to comment on the pr in the right place instead of the comments here

**Comment by hillslatt** [comment]
> we also need a pr on https://github.com/InteractiveAdvertisingBureau/openrtb/blob/main/extensions/community_extensions/segtax.md or adcom define segtax 8

There is a PR against AdCom to enumerate Content 3.1 as 9 (Ad Product 2.0 is 8)

**Comment by hillslatt** [comment]
> This continues one of the largest problems in openrtb, that cat, pagecat, bcats etc can only have one taxonomy. There is no upgrade path when a new taxonomy is released and you will be stuck on whatever the initial taxonomy is **forever**. Supply will never be able to stop sending the first taxonomy for fear of being misunderstood by non-adopters of the new taxonomy. Demand will never release support for a new taxonomy for lack of supply adoption.
> 
> This was solved in site.content.data and user.data like this: https://github.com/InteractiveAdvertisingBureau/openrtb/blob/main/extensions/community_extensions/segtax.md#example and i think we should repeat that pattern for genres.
> 
> ```
> "site": { 
>   "content": {
>     "data": [
>       {
> 	"name": "a-data-provider.com",
> 	"ext": {
> 	  "segtax": 8,
>           "taxtype": 'genre'
> 	},
> 	"segment": [
> 	  { "id": "1001" },
> 	  { "id": "1002" }
> 	]
>       }
>     }
>   }
> }
> ```

Fair enough, but I don't love using site.content.data. Would you be open to making gtax and genre elements of an object to pass multiple taxos as needed? We could potentially tackle doing the same for site/app.cattax for bcat and cat next year if it works.

**Comment by mike-chowla** [comment]
I 100% agree with @patmmccann that we should not repeat the mistakes of the past and create a spec that only allows one taxonomy to be sent in a request.  As an industry, we are stuck using IAB 1.0 taxonomy with cat, pagecat, bcats because there's no feasible migration path when it's not possible to send fields using both the old and new versions in a bid request.

**Comment by edandrea1** [comment]
Is there any proposed language to deal with `genre` and `genres` being passed in the same bid request?  I'm assuming `genre` will be listed as deprecated at some point, but should it be explicitly stated the `genres` takes precedence if both are included?  It's an edge case but I'd like to have clear guidance on how to handle this.

**Comment by alyssapyros** [comment]
we have a few other audio genres in our taxonomy that we worked with a popular audio provider to align on based on their top genre lists. have we considered adding Latin Music, Punk Music, Indie Rock, House Music, or Traditional Music? i do see 
Special Interest (Indie/Art House) as one but wanted to note that we broke them out based on popularity

**Comment by mhgifis** [comment]
Agreed, especially when considering there is an emerging push vs. pull – between structured taxonomies vs. more elastic tagging values.  Why put a new spec in place that is just like the old spec (even if w different values) and not support the LLM-era being rapidly adopted by supply, demand, supply chain intermediary and 3PD.

**Comment by Ryan-Dilworth** [comment]
Would the general guidance be to continue to pass genre name in`genre` and enumerated values in `genres`?  At least until there is broad adoption of `genres`?

**Comment by jimgaop1** [comment]
1. Could you pls provide guidance on whether to pass an array of strings or an array of IDs in the "genres" field?
2. A good amount of live TV programs are on news. Not seeing "news" as a genre in the CTV genre mapping list. would this be a good addition to the genres?
Thank you!

**Comment by hillslatt** [comment]
> I 100% agree with @patmmccann that we should not repeat the mistakes of the past and create a spec that only allows one taxonomy to be sent in a request. As an industry, we are stuck using IAB 1.0 taxonomy with cat, pagecat, bcats because there's no feasible migration path when it's not possible to send fields using both the old and new versions in a bid request.

hear you @mike-chowla - I think the guidance will be that exchange partners will be expected to hold these mappings, but we'll chat about in the next Prog Supply Chain meeting.

**Comment by hillslatt** [comment]
> Is there any proposed language to deal with `genre` and `genres` being passed in the same bid request? I'm assuming `genre` will be listed as deprecated at some point, but should it be explicitly stated the `genres` takes precedence if both are included? It's an edge case but I'd like to have clear guidance on how to handle this.

That's a fantastic call out @edandrea1 ! I'll add it to the discussion for next Prog Supply Chain meeting, but I'm super aligned with this.

**Comment by hillslatt** [comment]
> we have a few other audio genres in our taxonomy that we worked with a popular audio provider to align on based on their top genre lists. have we considered adding Latin Music, Punk Music, Indie Rock, House Music, or Traditional Music? i do see Special Interest (Indie/Art House) as one but wanted to note that we broke them out based on popularity

@alyssapyros Most of those rows are in Content Taxonomy 3.0 under Entertainment (starting at row 339) but I LOVE the idea of doing the some imp guidance specifically aimed at music to support this use case. I'm sure that'll also help us find some gaps that require new rows to the Taxonomy. Let's work with @katieshell and that group to figure out next steps.

**Comment by hillslatt** [comment]
> Would the general guidance be to continue to pass genre name in`genre` and enumerated values in `genres`? At least until there is broad adoption of `genres`?

@Ryan-Dilworth That would be my expectation, but it'd be up to implementers. Will add the question to the agenda for the next Prog Supply Chain meeting for the community to discuss.

**Comment by hillslatt** [comment]
> 1. Could you pls provide guidance on whether to pass an array of strings or an array of IDs in the "genres" field?
> 2. A good amount of live TV programs are on news. Not seeing "news" as a genre in the CTV genre mapping list. would this be a good addition to the genres?
>    Thank you!

@jimgaop1 - 

1. 'genres' is an array of IDs as enumerated in the subset of [Content 3.0 for CTV](https://github.com/InteractiveAdvertisingBureau/Taxonomies/blob/develop/Taxonomy%20Mappings/CTV%20Genre%20Mapping.tsv)
2. Great catch! The News ID is located in [Content Taxonomy 3.0 Descriptive Vectors](https://github.com/InteractiveAdvertisingBureau/Taxonomies/blob/develop/Content%20Taxonomies/Content%20Taxonomy%203.0%20Descriptive%20Vectors.tsv) (ID 1020).  @katieshell will add the row to the [CTV Genre List](https://github.com/InteractiveAdvertisingBureau/Taxonomies/blob/develop/Taxonomy%20Mappings/CTV%20Genre%20Mapping.tsv) as part of Public Comment feedback.
